### PR TITLE
Implemented "Go to Implementation" and "Peek Implementation" 

### LINKED ISSUE #37: Add Go To Implementation
Go to implementaiton is more useful in most cases than going to definition, is there a way to add to VScode

### PR DESCRIPTION
**Pull request by ivanz**
Closes #37

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,13 @@
+## 1.8.0 _(Not Yet Released)_
+
+### Code Actions
```
Comment (DustinCampbell) [inline CHANGELOG.md]: Four hashes please

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,13 @@
...
+
+* Support for "Go to Implementation" and "Peek Implementation". ([#37](https://github.com/OmniSharp/omnisharp-vscode/issues/37)) _(Contributed by [@ivanz](https://github.com/ivanz))_
+
+
 ## 1.7.0 (February 8, 2017)
 
-#### Syntax Hightlighting
+### Syntax Hightlighting
```
Comment (DustinCampbell) [inline CHANGELOG.md]: Please don't change this

File: src/features/implementationProvider.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import AbstractSupport from './abstractProvider';
+import { FindImplementationsRequest } from '../omnisharp/protocol';
+import * as serverUtils from '../omnisharp/utils';
+import { createRequest, toLocation } from '../omnisharp/typeConvertion';
+import { TextDocument, Position, CancellationToken, ImplementationProvider, ProviderResult, Definition } from 'vscode';
+
+export default class CSharpImplementationProvider extends AbstractSupport implements ImplementationProvider {
+    constructor(server) {
+        super(server);
+    }
```
Comment (DustinCampbell) [inline src/features/implementationProvider.ts]: Is this strictly needed if you're just calling super?

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,6 +1,13 @@
+## 1.8.0 _(Not Yet Released)_
+
+### Code Actions
+
+* Support for "Go to Implementation" and "Peek Implementation". ([#37](https://github.com/OmniSharp/omnisharp-vscode/issues/37)) _(Contributed by [@ivanz](https://github.com/ivanz))_
```
Comment (DustinCampbell) [inline CHANGELOG.md]: autobiographical. :smile:

File: CHANGELOG.md (hunk 1 above)
Comment (DustinCampbell) [inline CHANGELOG.md]: Also, I wouldn't put this under "Code Actions". Those are for code fixes and refactorings that appear in the light bulb menu. I'd actually give this it's own subheading: "Go to Implementation".

**Comment by DustinCampbell** [comment]
Fantastic @ivanz! :smile;

I'll be back in a couple of hours. If you want to make those small changes to the changelog I requested, I'll merge when I'm back.

File: CHANGELOG.md (hunk 4 above)
Comment (filipw) [inline CHANGELOG.md]: 😅

**Comment by ivanz** [comment]
Updated :)

**Comment by DustinCampbell** [comment]
Thanks again @ivanz. You rock!

**Comment by jchannon** [comment]
Awesome stuff!? Is there a CI build to try this out?

**Comment by DustinCampbell** [comment]
No, we don't push CI builds. I'll need to create a beta release, which I was going to take a look at shortly. I need to get an omnisharp-roslyn build first to ensure that other recently-fixed issues are included in the first 1.8 beta build.

**Comment by jchannon** [comment]
Cool. No worries

**Pull request merged**